artemis-server/src/test/java/org/apache/activemq/artemis/core/config/impl/ConfigurationImplTest.java

```diff
@@ -1554,6 +1555,48 @@ public void testSystemPropKeyReplacement() throws Exception {
+      Properties insertionOrderedProperties = new ConfigurationImpl.InsertionOrderedProperties();
+      insertionOrderedProperties.put("storeConfiguration", "DATABASE");
+      insertionOrderedProperties.put("storeConfiguration.largeMessageTableName", "lmtn");
+      insertionOrderedProperties.put("storeConfiguration.messageTableName", "mtn");
+      insertionOrderedProperties.put("storeConfiguration.bindingsTableName", "btn");
+      insertionOrderedProperties.put("storeConfiguration.dataSourceClassName", "dscn");
+      insertionOrderedProperties.put("storeConfiguration.nodeManagerStoreTableName", "nmtn");
+      insertionOrderedProperties.put("storeConfiguration.pageStoreTableName", "pstn");
+      insertionOrderedProperties.put("storeConfiguration.jdbcAllowedTimeDiff", 123);
+      insertionOrderedProperties.put("storeConfiguration.jdbcConnectionUrl", "url");
+      insertionOrderedProperties.put("storeConfiguration.jdbcDriverClassName", "dcn");
+      insertionOrderedProperties.put("storeConfiguration.jdbcJournalSyncPeriodMillis", 456);
+      insertionOrderedProperties.put("storeConfiguration.jdbcLockAcquisitionTimeoutMillis", 789);
+      insertionOrderedProperties.put("storeConfiguration.jdbcLockExpirationMillis", 321);
+      insertionOrderedProperties.put("storeConfiguration.jdbcLockRenewPeriodMillis", 654);
+      insertionOrderedProperties.put("storeConfiguration.jdbcNetworkTimeout", 987);
+      insertionOrderedProperties.put("storeConfiguration.jdbcPassword", "pass");
+      insertionOrderedProperties.put("storeConfiguration.jdbcUser", "user");
+      configuration.parsePrefixedProperties(insertionOrderedProperties, null);
+      Assert.assertTrue(configuration.getStoreConfiguration() instanceof DatabaseStorageConfiguration);
+      DatabaseStorageConfiguration dsc = (DatabaseStorageConfiguration) configuration.getStoreConfiguration();
+      Assert.assertEquals(dsc.getLargeMessageTableName(), "lmtn");
+      Assert.assertEquals(dsc.getMessageTableName(), "mtn");
+      Assert.assertEquals(dsc.getBindingsTableName(), "btn");
+      Assert.assertEquals(dsc.getDataSourceClassName(), "dscn");
+      Assert.assertEquals(dsc.getJdbcAllowedTimeDiff(), 123);
+      Assert.assertEquals(dsc.getJdbcConnectionUrl(), "url");
+      Assert.assertEquals(dsc.getJdbcDriverClassName(), "dcn");
+      Assert.assertEquals(dsc.getJdbcJournalSyncPeriodMillis(), 456);
+      Assert.assertEquals(dsc.getJdbcLockAcquisitionTimeoutMillis(), 789);
+      Assert.assertEquals(dsc.getJdbcLockExpirationMillis(), 321);
+      Assert.assertEquals(dsc.getJdbcLockRenewPeriodMillis(), 654);
+      Assert.assertEquals(dsc.getJdbcNetworkTimeout(), 987);
+      Assert.assertEquals(dsc.getJdbcPassword(), "pass");
+      Assert.assertEquals(dsc.getJdbcUser(), "user");
+      Assert.assertEquals(dsc.getNodeManagerStoreTableName(), "nmtn");
+      Assert.assertEquals(dsc.getPageStoreTableName(), "pstn");
+   }
+
    @Test
```

Comment:
a negative test where the value is wrong would be great. to see what error comes out. the error should be in the Status.